### PR TITLE
fix: derive snapshot task name suffix from schedule as a fallback (#55)

### LINKED ISSUE #55: [Feature] Better naming for Snapshot tasks
## Is your feature request related to a problem? Please describe.

I have multiple snapshot tasks for a dataset. They run hourly, daily and weekly. The issue is that the entities from the integration are all named the after the dataset alone. And the id of duplicated tasks are named with a subfix "_2", "_3" and so on. It makes managing them very difficult for automation and overview.

## Describe the solution you'd like

I know it is caused by the API of the TrueNAS. However, I have added a subfix to the naming scheme of the snapshots to the tasks in TrueNAS with "_daily" and "_weekly", e.g. "auto-%Y-%m-%d_%H-%M_daily" and "auto-%Y-%m-%d_%H-%M_weekly"

I wonder if it is possible somehow utilizing the naming scheme of the TrueNAS snapshot tasks in order to produce more decipherable ids and names of the entities of snapshot tasks for human consumption.

As you know, I have been testing and migrating the integration from repositories and versions moderately frequently. It takes me ample of time to rename the entities on each test and update. It gets worsen with the additional of the control entities which double the number of the snapshot task entities. Multiple instances of Home Assistants run by me is not helping either.

## Describe alternatives you've considered

Just keep the current implementation and increases the effort of testing and migrating between repositories and versions.

### PR DESCRIPTION
**Pull request by kayl-codes**
## Proposed change
Follow-up to #59 for issue #55. Users with several same-dataset snapshot tasks that all still use the plain default `naming_schema` (no literal suffix) still get colliding names, since the previous fix only disambiguates when `naming_schema` carries custom literal text.

This adds a fallback: parse the task's cron `schedule` field (now captured by `get_snapshottask`) and, when `naming_schema` gives no suffix, guess a `Hourly`/`Daily`/`Weekly`/`Monthly` label from it, matching TrueNAS's known periodic-snapshot-task schedule presets (`dom` set → Monthly, `dow` set → Weekly, `hour` set → Daily, else Hourly). `naming_schema`'s literal suffix still wins when present.

Also bumps `_LOG_PAYLOAD_LIMIT` (500 → 5000 chars) so debug-log captures of `pool.snapshottask.query` aren't truncated after the first list entry — that's what limited us to a single confirmed example (Hourly) from the reporter's data.

**Caveat:** the Daily/Weekly/Monthly branches are a best-effort guess based on TrueNAS's known presets, not yet confirmed against real non-hourly schedule data (see the reporter's debug log, which was truncated before this fix). Shipping as a prerelease for confirmation before it goes stable.

Closes #55 (pending reporter confirmation).

## Type of change
- [x] Bugfix
- [ ] New feature
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation

## Additional information
Follow-up to #59 (v2.6.0-rc1). Will ship as v2.6.0-rc2 for @janusn to verify before stable v2.6.0.

## Checklist
- [x] The code change is tested and works locally.
- [x] The code has been formatted and linted using Ruff.
- [x] Tests have been added to verify that the new code works.
- [ ] Documentation added/updated if required.
- [x] Tested against the latest Home Assistant version.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Improve TrueNAS snapshot task sensor naming to avoid collisions by deriving a cadence suffix from either the naming schema or the task schedule, and ensure schedule data is captured and logged sufficiently for debugging.

Bug Fixes:
- Avoid entity name collisions for same-dataset snapshot tasks that use the default naming_schema by falling back to an Hourly/Daily/Weekly/Monthly suffix inferred from the task schedule.
- Ensure snapshot task schedule data is stored by the coordinator, defaulting to an empty dict when missing, so sensors can reliably derive names.

Enhancements:
- Add robust schedule classification logic that only labels tasks when their cron schedule matches known TrueNAS presets, avoiding misleading cadence guesses.
- Increase API payload debug log limit to capture more of large responses such as pool.snapshottask.query for better diagnostics.

Tests:
- Extend snapshot task sensor tests to cover schedule-based naming, precedence of naming_schema suffix over schedule, and various cron schedule edge cases.
- Update coordinator tests to verify that snapshot task schedules are parsed and defaulted correctly.

### CODE REVIEW

File: tests/test_sensor.py [hunk 1]
```diff
@@ -485,6 +485,67 @@ def test_snapshottask_name_uses_naming_schema_suffix(
     assert sensor.name == expected_name
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply tests/test_sensor.py:485]: Added -- both a non-preset schedule (`hour: "*/2"`, wildcarded but not the literal `"*"`) and a dom+dow-both-pinned case are now covered in `test_snapshottask_name_falls_back_to_schedule`. This also surfaced that the original `hour not in ("*", None)` check would have misclassified `"*/2"` as a fixed hour (-> Daily) -- tightened `_schedule_suffix` to require the field contain no wildcard character at all, not just differ from the literal `"*"`, per your overall comment about guarding non-preset schedules.

File: tests/test_coordinator.py [hunk 2]
```diff
@@ -3050,9 +3050,13 @@ async def test_get_snapshottask_parses_when_monitored() -> None:
...
     coord.config_entry.options = {CONF_MONITORED_GROUPS: [MONITOR_GROUP_SNAPSHOTS]}
     coord.api = MagicMock()
-    coord.api.query = AsyncMock(return_value=[{"id": 1, "dataset": "tank/data"}])
+    schedule = {"minute": "0", "hour": "*", "dom": "*", "month": "*", "dow": "*"}
+    coord.api.query = AsyncMock(
+        return_value=[{"id": 1, "dataset": "tank/data", "schedule": schedule}]
+    )
     await coord.get_snapshottask()
     assert 1 in coord.ds["snapshottask"]
+    assert coord.ds["snapshottask"][1]["schedule"] == schedule
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply tests/test_coordinator.py:3059]: Added `test_get_snapshottask_defaults_schedule_when_missing`, asserting the task falls back to the vals spec's `{}` default when the API response omits `schedule` entirely (older TrueNAS versions or atypical responses).

**Comment by kayl-codes** [comment]
Re the overall comment on `_LOG_PAYLOAD_LIMIT`: I'll leave it a fixed constant rather than a user-facing setting. It only bounds the length of one internal debug-log line -- it's not something a typical user would tune, and exposing it as an options-flow setting would add config-flow/strings.json surface for a one-off debugging aid. 5000 chars comfortably covers the snapshot-task lists seen so far; if a future payload needs more, bumping the constant is a one-line change.

Re the `_schedule_suffix` precedence comment: addressed by tightening the heuristic (see reply below) plus an expanded docstring documenting the dom-over-dow precedence for non-preset schedules.

Individual comments addressed in 8f99185.

**Comment by kayl-codes** [comment]
@sourcery-ai review

**Comment by kayl-codes** [comment]
Addressed both overall comments in b13b159:

- `_is_pinned_schedule_field` now requires a single fixed number (`str.isdigit()`), so ranges (`1-5`) and lists (`1,15`) no longer count as pinned.
- More importantly, if *any* of dom/dow/hour/month is neither wildcarded nor a single fixed number, the whole schedule is now left unclassified -- previously a non-preset dow/dom could still fall through and get mislabeled off the hour field. Added tests for both the range/list case and a pinned-month case.
- Pinned `month` is explicitly guarded (never occurs in any of the four known presets), also documented in the docstring.

**Comment by kayl-codes** [comment]
@sourcery-ai review

**Comment by kayl-codes** [comment]
Addressed the third round of review feedback in f38650f:

- `_is_pinned_schedule_field` now accepts plain `int` values as pinned (not just digit-only strings), guarding against `bool` (since `isinstance(True, int)` is `True` in Python) so a future/unexpected API response using numbers instead of strings still classifies correctly.
- `_schedule_suffix` now includes `minute` in the shape-validation check, so a step like `minute: "*/5"` (e.g. "every 5 minutes") correctly disqualifies the schedule from any preset, even if `hour`/`dom`/`dow`/`month` otherwise look like the Hourly shape. `minute` is still excluded from the dom -> dow -> hour classification itself, since every preset pins it the same way.
- The final Hourly check now uses `_is_wildcard_schedule_field(hour)` instead of the literal `hour == "*"`, so a missing/`None` `hour` key is treated the same as an explicit wildcard, consistent with how every other field in this function is handled.
- Added a guard for an entirely empty `schedule` dict: without it, treating `None` as wildcard for a fully-empty dict would fall through to labeling it Hourly (every field "missing" = every field "wildcard"), which isn't a real signal. `{}` (and non-dict values) now stay unclassified as before.
- Added test cases: a `minute: "*/5"` step, plain-int schedule values, and a missing `hour` key.

All checks green locally: 808 tests passed, ruff clean, mypy clean, bandit clean (same pre-existing `nosec` warning).

**Comment by kayl-codes** [comment]
@sourcery-ai review

**Comment by kayl-codes** [comment]
Addressed the round-4 review feedback in 7abfe78:

- **Centralized cron field names**: extracted the `("minute", "dom", "dow", "hour", "month")` tuple into a module-level `_SCHEDULE_FIELDS` constant, used at the single call site that reads them off the `schedule` dict.
- **Simplified the trailing hour check**: the `any(...)` shape guard already guarantees every field is pinned-or-wildcard, and by the time execution reaches the end of the dom/dow/hour chain, `hour` didn't match the pinned branch above -- so it's provably wildcard there. Replaced the redundant `if _is_wildcard_schedule_field(hour): ...; return None` with a comment stating the invariant, plus the new minute check below.
- **Fully wildcard schedule no longer mislabeled Hourly** (individual comment 1): a genuine Hourly preset always pins `minute` to the run time within the hour. If `minute` is *also* wildcard, that's "every minute", not hourly, so it's now left unclassified (`return None`) rather than guessed at. Added a test case (`{"minute": "*", "hour": "*", "dom": "*", "month": "*", "dow": "*"}` -> plain dataset-only name) to cover this.

Verified: `ruff check`/`ruff format --check` clean, `mypy` clean (21 files), `pytest` 809/809 passed (up from 808), `bandit` clean (same pre-existing nosec on `const.py:100`, no new findings).

**Comment by kayl-codes** [comment]
@sourcery-ai review
